beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -353,34 +353,71 @@ impl<AppReqId: ReqId, TSpec: EthSpec> Network<AppReqId, TSpec> {
             }
         };
 
-        let (swarm, bandwidth) = {
-            // Set up the transport - tcp/ws with noise and mplex
-            let (transport, bandwidth) = build_transport(local_keypair.clone())
-                .map_err(|e| format!("Failed to build transport: {:?}", e))?;
-
-            // use the executor for libp2p
-            struct Executor(task_executor::TaskExecutor);
-            impl libp2p::swarm::Executor for Executor {
-                fn exec(&self, f: Pin<Box<dyn futures::Future<Output = ()> + Send>>) {
-                    self.0.spawn(f, "libp2p");
-                }
+        struct Executor(task_executor::TaskExecutor);
```

Comment:
Why does lighthouse need a custom executor?

Reply:
Provides additional features. See https://github.com/sigp/lighthouse/blob/stable/common/task_executor/src/lib.rs.

Reply:
Should we support this usecase better in the new `SwarmBuilder`?

Reply:
Yeah our executor allows us to exit tasks a bit easier and adds some metrics so we can see what's going on.